antarest/study/service.py

```diff
@@ -1962,3 +1981,15 @@ def update_matrix(
         assert_permission(params.user, study, StudyPermissionType.WRITE)
         self._assert_study_unarchived(study)
         self.matrix_manager.update_matrix(study, path, slices, operation)
+
+    def check_and_update_all_study_versions_in_database(
+        self, params: RequestParameters
+    ) -> None:
+        if params.user and not params.user.is_site_admin():
+            logger.error(f"User {params.user.id} is not site admin")
+            raise ForbiddenError("Only site admins can update study versions")
+        studies = self.repository.get_all()
+        for study in studies:
+            self.storage_service.get_storage(
```

Comment:
here you could filter only RawStudy or just studies that are not managed (there is a util function is_managed somewhere)